### PR TITLE
Elastic-agent: add documentation about running agent on managed kubenetes environmets, need to adjust resource configuration

### PR DESCRIPTION
**Pull request by tetianakravchenko**
Extend elastic-agent documentation:
- about running agent on managed kubenetes environments, that not all metrics and logs can be available
- add note, that container resources might be needed to adjust

Signed-off-by: Tetiana Kravchenko <tetiana.kravchenko@elastic.co>

### CODE REVIEW

**Review by MichaelKatsoulis — APPROVED** [review]
LGTM!

**Review by ChrsMark — APPROVED** [review]

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc [hunk 1]
```diff
@@ -147,6 +147,9 @@ elastic-agent-hrjbg   1/1     Running   0          12m
 elastic-agent-olpsd   1/1     Running   0          12m
 ------------------------------------------------
 
+NOTE: you might need to adjust https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/[resource limits] of elastic-agent container
+in the `elastic-agent-managed-kubernetes.yaml` manifest: container resources usage depends on number of datastreams
+and the environment size.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc:152]:
```suggestion
NOTE: You might need to adjust https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/[resource limits] of the elastic-agent container
in the `elastic-agent-managed-kubernetes.yaml` manifest. Container resource usage depends on the number of datastreams
and the environment size.
```

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc [hunk 2]
```diff
@@ -159,4 +162,12 @@ The size and the number of nodes in a Kubernetes cluster can be fairly large at
 and in such cases the Pod that will be collecting cluster level metrics might face performance
 issues due to resources limitations. In this case users might consider to avoid using the
 leader election strategy and instead run a dedicated, standalone {agent} instance using
-a Deployment in addition to the DaemonSet.
\ No newline at end of file
+a Deployment in addition to the DaemonSet.
+
+[discrete]
+== Deploying {agent} to managed kubenetes environment
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc:168]:
```suggestion
== Deploying {agent} to managed Kubernetes environment
```

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc [hunk 3]
```diff
@@ -159,4 +162,12 @@ The size and the number of nodes in a Kubernetes cluster can be fairly large at
...
+
+On managed kubernetes solutions such as AKS, GKE or EKS, elastic-agent has no access to collect metrics from the [kubernetes control plane](https://kubernetes.io/docs/concepts/overview/components/#control-plane-components)
+components, like `kube-scheduler` and `kube-controller-manager`, that are scheduled on kubernetes master nodes.
+
+Audit logs are available only on kubernetes master nodes as well, hense can not be collected by elastic-agent.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc:173]:
```suggestion
On managed Kubernetes solutions, such as AKS, GKE or EKS, {agent} has no access to collect metrics from the [Kubernetes control plane](https://kubernetes.io/docs/concepts/overview/components/#control-plane-components)
components, like `kube-scheduler` and `kube-controller-manager`, that are scheduled on Kubernetes master nodes.

Audit logs are available only on Kubernetes master nodes as well and hence cannot be collected by {agent}.
```

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc [hunk 4]
```diff
@@ -124,6 +124,10 @@ elastic-agent-fj2z9             1/1     Running   0          81m
 elastic-agent-hs4pb             1/1     Running   0          81m
 ------------------------------------------------
 
+NOTE: you might need to adjust https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/[resource limits] of elastic-agent container
+in `elastic-agent-standalone-kubernetes.yaml` manifest: container resources usage depends on number of datastreams
+and the environment size.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc:129]:
```suggestion
NOTE: You might need to adjust https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/[resource limits] of the elastic-agent container
in the `elastic-agent-standalone-kubernetes.yaml` manifest. Container resource usage depends on the number of datastreams
and the environment size.
```

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc [hunk 5]
```diff
@@ -264,3 +268,11 @@ and in such cases the Pod that will be collecting cluster level metrics might fa
 issues due to resources limitations. In this case users might consider to avoid using the
 leader election strategy and instead run a dedicated, standalone {agent} instance using
 a Deployment in addition to the DaemonSet.
+
+[discrete]
+== Deploying {agent} to managed kubenetes environment
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc:273]:
```suggestion
== Deploying {agent} to managed Kubernetes environment
```

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc [hunk 6]
```diff
@@ -264,3 +268,11 @@ and in such cases the Pod that will be collecting cluster level metrics might fa
...
+
+On managed kubernetes solutions such as AKS, GKE or EKS, elastic-agent has no access to collect metrics from the [kubernetes control plane](https://kubernetes.io/docs/concepts/overview/components/#control-plane-components)
+components, like `kube-scheduler` and `kube-controller-manager`, that are scheduled on kubernetes master nodes.
+
+Audit logs are available only on kubernetes master nodes as well, hense can not be collected by elastic-agent.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc:278]:
```suggestion
On managed Kubernetes solutions, such as AKS, GKE or EKS, {agent} has no access to collect metrics from the [Kubernetes control plane](https://kubernetes.io/docs/concepts/overview/components/#control-plane-components)
components, like `kube-scheduler` and `kube-controller-manager`, that are scheduled on Kubernetes master nodes.

Audit logs are available only on Kubernetes master nodes as well, and hence cannot be collected by {agent}.
```

**Review by dedemorton — COMMENTED** [review]
Just some editorial changes. Thanks for your contribution to the docs!

**Comment by tetianakravchenko** [comment]
@dedemorton thank you for the review! I've merged all suggestions

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc [hunk 7]
```diff
@@ -159,4 +162,12 @@ The size and the number of nodes in a Kubernetes cluster can be fairly large at
...
 issues due to resources limitations. In this case users might consider to avoid using the
 leader election strategy and instead run a dedicated, standalone {agent} instance using
-a Deployment in addition to the DaemonSet.
\ No newline at end of file
+a Deployment in addition to the DaemonSet.
+
+[discrete]
+== Deploying {agent} to managed Kubernetes environment
+
+On managed Kubernetes solutions, such as AKS, GKE or EKS, {agent} has no access to collect metrics from the [Kubernetes control plane](https://kubernetes.io/docs/concepts/overview/components/#control-plane-components)
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc:170]:
```suggestion
On managed Kubernetes solutions, such as AKS, GKE or EKS, {agent} has no access to collect metrics from the https://kubernetes.io/docs/concepts/overview/components/#control-plane-components[Kubernetes control plane]
```

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc [hunk 8]
```diff
@@ -264,3 +268,11 @@ and in such cases the Pod that will be collecting cluster level metrics might fa
 issues due to resources limitations. In this case users might consider to avoid using the
 leader election strategy and instead run a dedicated, standalone {agent} instance using
 a Deployment in addition to the DaemonSet.
+
+[discrete]
+== Deploying {agent} to managed Kubernetes environment
+
+On managed Kubernetes solutions, such as AKS, GKE or EKS, {agent} has no access to collect metrics from the [Kubernetes control plane](https://kubernetes.io/docs/concepts/overview/components/#control-plane-components)
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc:275]:
```suggestion
On managed Kubernetes solutions, such as AKS, GKE or EKS, {agent} has no access to collect metrics from the https://kubernetes.io/docs/concepts/overview/components/#control-plane-components[Kubernetes control plane]
```

**Review by dedemorton — COMMENTED** [review]
My apologies for missing this before, but the syntax for links here is wrong (it's markdown)

**Review by dedemorton — APPROVED** [review]
LGTM!

**Pull request merged**